### ISSUE TITLE
PC - Lecture 1

### ISSUE THREAD

**Issue by AranScope** [AranScope/y2-lectures#2]
An ethical interlude --

You are a computer system manager. One of your employees is sick and another employee requests; on behalf of the entire development team, that all of their files are copied to the development team. These files are required to complete work towards an urgent deadline. What do you do?

Well, the team should have been using a version control and code sharing platform. By having project files stored in only one place
## Module Overview

Professional computing covers the ethical, social, legal and professional issues around the working world of computing.
## Assesment

50% Exam (May)

50% CA
- 15% CV
- 15% Group presentation
- 20% Essay
## Timetable
- Week 1: One lecture in the 2-3pm slot
- Week 2: Lecture 2-3pm, Lecture 5-6pm
- Week 3: CV Lecture 2-3pm, Careers talk 4-6pm
- Week 4: No lectures
- Week 5: Lecture 2-3pm, Lecture 4-5pm
- Weeks 6-11: Lecture 2-3pm, Presentations 4-6pm

Another ethical interlude -- Pull the lever!

You are the driver of a computer controlled trolley, lack of funding has meant that that the trolley hasn't been properly tested and has a fatal flaw -- when you brake, you actually accelerate!

You come to a split on the road, the left side has a group of 5 workmen that will certainly be killed if you choose that route, the right, a lone workman who will again be killed. What do you do?
## Presentations

The audience will be split into two groups:
- A - 4-5pm Tuesdays
- B - 5-6pm Tuesdays

Groups will consist of 5-6 students with randomly assigned topics (talks to be given in week 3). Make sure to attend other groups talks -- and your own!!.
## Lateness and Plagiarism
- Late submitted work = 5% penalty per working day
- Plagiarism = Senior tutor investigation
## Ethical issues in CS
- Should backdoors be given to law enforcement agencies to help prevent crimes?
- Whistleblowers, is there good reason to leak privileged information to the public in some cases?
- Self-driving cars, again the 'pull the lever' dilemma
- Automation in manufacturing, replacing low-skilled workers
- Should deepmind be allowed access to private medical records?
## Referencing
- You will learn how to use academic references in technical writing (Harvard style)
- You will learn good general academic scholarship
## Other tools

You are expected to use LaTeX for technical report writing. LaTeX is a markup language for typesetting. Lingua franca for scientific writing (and mathematics). You should also use LaTeX for your final year projects.

You should use Subversion to manage your reports, it is an industry and research standard, designed for coding but also works well for multiple author writing.

Overleaf is a nice google doc like editor for LaTeX.

-- an aside about what it means to be a professional and professional bodies --
## BCS

The IT industry in the UK is largely unregulated, but we do have a professional body -- BCS, they do not have regulatory powers but do: 
- Accredit degree programmes
- Advise the government on IT related issues
- Disseminate good practice and attempt to define 'professional behaviour'

Some argue the term 'software engineer' should be restricted, the following are illustrative of this:
- London ambulance scheduling service
- Therac-25 disaster

To summarise our ethical dilemmas, ethics is difficult! You have a set of information but your ethics and behaviour are based on your experience, morals etc. and there is no 'black and white' decision process.
